Modules/_sqlite/connection.c

```diff
@@ -440,40 +442,56 @@ remove_callbacks(sqlite3 *db)
 static void
 connection_close(pysqlite_Connection *self)
 {
-    if (self->db) {
-        if (self->autocommit == AUTOCOMMIT_DISABLED &&
-            !sqlite3_get_autocommit(self->db))
-        {
-            /* If close is implicitly called as a result of interpreter
-             * tear-down, we must not call back into Python. */
-            if (_Py_IsInterpreterFinalizing(PyInterpreterState_Get())) {
-                remove_callbacks(self->db);
-            }
-            (void)connection_exec_stmt(self, "ROLLBACK");
+    assert(self->db);
+    if (self->autocommit == AUTOCOMMIT_DISABLED &&
+        !sqlite3_get_autocommit(self->db))
+    {
+        /* If close is implicitly called as a result of interpreter
+         * tear-down, we must not call back into Python. */
+        if (_Py_IsInterpreterFinalizing(PyInterpreterState_Get())) {
+            remove_callbacks(self->db);
         }
+        (void)connection_exec_stmt(self, "ROLLBACK");
```

Comment:
Ignoring the exception here is a bug.

connection_finalize() clears any exception with PyErr_SetRaisedException(), but pysqlite_connection_close_impl() must not ignore silently error, since here we are talking about a raised Python exception! The function must report if an exception was raised. Then the caller is free to ignore it or not.

I suggest to continue ignoring it in finalize, but then write a separated PR to *log* the "unraisable exception".

Reply:
Ouch! And there is at least one more case where the exception is ignored. Let's fix those in a separate PR.

Reply:
See gh-108084